src/components/BulkOperations/BulkOperations.test.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {
+  render, screen, fireEvent, within, waitFor,
+} from '@testing-library/react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import BulkOperations from './index';
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+import '@testing-library/jest-dom';
+
+const mockedHistoricalTasks = [
+  {
+    id: 1,
+    uploaded_by: 'edx',
+    task_type: 'course_create',
+    status: 'completed',
+    created: '2025-05-05T00:00:00Z',
+    modified: '2025-05-05T00:00:00Z',
+    csv_file: 'https://foo.com/bar.csv',
+  },
+  {
+    id: 2,
+    uploaded_by: 'edx',
+    task_type: 'course_rerun',
+    status: 'failed',
+    created: '2025-05-05T00:00:00Z',
+    modified: '2025-05-05T00:00:00Z',
+    csv_file: 'https://foo.com/baz.csv',
+  },
+];
+
+const createdTask = {
+  ...mockedHistoricalTasks[0],
+  id: 3,
+  status: 'pending',
+  csv_file: 'https://foo.com/hello.csv',
+};
+
+describe('BulkOperationsPage', () => {
+  let post;
+  beforeAll(() => {
+    const get = jest.spyOn(DiscoveryDataApiService, 'fetchBulkOperations');
+    post = jest.spyOn(DiscoveryDataApiService, 'createBulkOperation');
+    get.mockResolvedValue({ data: { results: mockedHistoricalTasks } });
+  });
+
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('initial render', async () => {
+    render(
+      <IntlProvider locale="en">
+        <BulkOperations />
+      </IntlProvider>,
```

Comment:
we can move this to beforeAll

Reply:
moved to `beforeEach`. I feel like `beforeAll` might give us issues, but I haven't tried it.